### PR TITLE
compile with setuptools

### PR DESCRIPTION
**Pull request by aleksandra-tarkowska**
## This PR

is created on top of https://github.com/ome/omero-figure/pull/178

It removes compiled JS files from the repo and allows to compile with setuptools. This will simplify the release process and testing

NOTE: this PR requires node and grunt installed on CI it should be excluded until this solution is accepted

## To test this PR:

install prerequisites node and grunt

```
$ mkdir test
$ cd test
$ virtualenv venvweb
$ source venvweb/bin/activate
(venvweb) $ pip install omego 
(venvweb) $ omego download --ice 3.6 --branch OMERO-DEV-merge-build py
(venvweb) $ mv OMERO.py[currentversion] OMERO.py
(venvweb) $ pip install -r OMERO.py/share/web/requirements-py27-trial.txt

(venvweb) $ pip install git+git://github.com/aleksandra-tarkowska/omero-figure.git@install#egg=omero-figure -vvv
(venvweb) $ OMERO.py/bin/omero config append omero.web.apps '"figure"'

# if you dont have local server set up eel
(venvweb) $ OMERO.py/bin/config set omero.web.server_list '[["eel.o.org", 4064, "eel"]]'

(venvweb) $ OMERO.py/bin/omero config set omero.web.application_server development
(venvweb) $ OMERO.py/bin/omero config set omero.web.debug True
(venvweb) $ OMERO.py/bin/omero web start
```

see travis output as well

```
$ python setup.py sdist install
running sdist
running grunt
running npm_install
npm install
npm WARN package.json omero-figure@2.0.1 license should be a valid SPDX license expression
grunt jst
Running "jst:compile" (jst) task
File omero_figure/static/figure/templates.js created.
Done, without errors.
grunt concat
Running "concat:js" (concat) task
File omero_figure/static/figure/figure.js created.
Done, without errors.
grunt jshint
Running "jshint:all" (jshint) task
>> 18 files lint free.
Done, without errors.
...
```

cc @jburel @waxenegger

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -44,6 +47,75 @@ def read_file(fname, content_type=None):
 LICENSE = "AGPLv3"
 HOMEPAGE = "https://github.com/ome/omero-figure"
 
+
+cmdclass = {}
+
+
+class NpmInstall(Command):
```
Comment (aleksandra-tarkowska) [inline setup.py]: I am not convince if we need this step

**Comment by aleksandra-tarkowska** [comment]
@waxenegger would https://github.com/ome/omero-figure/pull/188/commits/6bcf513dda5c07240c864dbd42e551d6ee5b052f work for you?

**Comment by jburel** [comment]
Thanks @aleksandra-tarkowska: let's chat tomorrow morning. 
I will try that this evening if I can

File: setup.py (hunk 1 above)
Comment (jburel) [inline setup.py]: I don't think this is required. 
I have it working w/o that step for iviewer.

Comment (aleksandra-tarkowska) [inline setup.py]: Removed

**Comment by aleksandra-tarkowska** [comment]
> for iviewer: only node will be needed. but not having node on CI is the same problem. I was looking into installing in virtualenv so different option possible

New role :)

**Comment by jburel** [comment]
in theory I agree but from what I understood ``ome-web-staging`` is not "ansible"ized (not sure that's a word!!)

**Comment by aleksandra-tarkowska** [comment]
Web are deployed on spiders. I think what we need really is node role added to https://github.com/openmicroscopy/infrastructure/blob/master/ansible/ci-deployment.yml#L30. Or you have your own CI nodes for viewer?

**Comment by jburel** [comment]
ok let's discuss tomorrow what's available, not planning to have specific nodes for viewer

**Comment by aleksandra-tarkowska** [comment]
I haven't tried releasing yet, like:

```
python setup.py register -r pypitest
python setup.py sdist upload -r pypitest
```

**Comment by jburel** [comment]
I think that's a good start already ;-)

**Comment by jburel** [comment]
corresponding iviewer PR https://github.com/ome/omero-iviewer/pull/2

**Comment by jburel** [comment]
you might be able to install grunt as a first command

**Comment by waxenegger** [comment]
@aleksandra-tarkowska sorry, you caught me a bit "out-of-context". if I understand correctly you are now working on the preferred way for pypi deployments/installs? Was the question in regards to figue or the iviewer?

**Comment by aleksandra-tarkowska** [comment]
@waxenegger would you like to have a look first in https://docs.python.org/2/distutils/packageindex.html ?

**Comment by aleksandra-tarkowska** [comment]
> you might be able to install grunt as a first command

@jburel perhaps you know more. I understood that grunt cli requires npm http://gruntjs.com/getting-started, this will face permission issues.

**Comment by jburel** [comment]
#178 has now been merged

**Comment by jburel** [comment]
@waxenegger: 
few things we need to achieve:
 * one is having the app "pip"installable. This is working
 * for app like figure or iviewer which required the code to be compiled. We don't want the compiled code in the repo (iviewer does not, but figure does) and we need to be able to easily  install the daily merge branch from snoopy i.e. ``pip install git+git: etc.``. One job will merge the opened PR from figure for example, then another one will "install". We will need node.js installed on the CI node but that's not part of this repo

**Comment by aleksandra-tarkowska** [comment]
@waxenegger See https://trello.com/c/ov0DR0VL/2605-nodejs-on-ci

**Comment by jburel** [comment]
@aleksandra-tarkowska:
I think you should re-introduce the ``npm install`` step
``
(pip-figure) ls30569:Desktop jmarie$ pip install -e git+git@github.com:jburel/omero-figure.git@install#egg=omero-figure
Obtaining omero-figure from git+git@github.com:jburel/omero-figure.git@install#egg=omero-figure
  Updating ./pip-figure/src/omero-figure clone (to install)
Installing collected packages: omero-figure
  Running setup.py develop for omero-figure
Successfully installed omero-figure-2.0.1
``

iviewer has the ``npm install`` step in https://github.com/ome/omero-iviewer/blob/master/prepare_build.sh

**Comment by jburel** [comment]
https://ci.openmicroscopy.org/view/Web/job/FIGURE-merge/325/ is green

excluding for now
I will prepare spiders and open PR

--exclude

**Comment by jburel** [comment]
Removing exclude label

**Comment by will-moore** [comment]
Do we need to delete ```omero_figure/static/figure/figure.js``` and ```templates.js``` from the repo (now that they're ignored).

**Comment by jburel** [comment]
https://ci.openmicroscopy.org/view/Web/job/WEB-DEV-trial-deploy/166/ is green

**Comment by jburel** [comment]
Tested also on testpypi to double-checked 
https://testpypi.python.org/pypi?:action=display&name=omero-figure&version=2.0.19
Everything works as expected
Merging

**Pull request merged**